src/core/fem/src/discretization/4C_fem_discretization_utils.cpp

```diff
@@ -121,4 +121,35 @@ void Core::FE::Utils::do_initial_field(const Core::Utils::FunctionManager& funct
   }
 }
 
+/*----------------------------------------------------------------------------*
+ *----------------------------------------------------------------------------*/
+Core::LinAlg::MultiVector<double> Core::FE::Utils::average_element_to_nodal_values(
+    const Discretization& discretization,
+    const Core::LinAlg::MultiVector<double>& element_multi_vector)
+{
+  Core::LinAlg::MultiVector<double> nodal_material_vector = Core::LinAlg::MultiVector<double>(
+      *discretization.node_row_map(), element_multi_vector.NumVectors(), true);
```

Comment:
I know, that you will use this to compute a "nodal material vector", but this generic function might be used for other things. Maybe just `nodal_vector`?
```suggestion
  Core::LinAlg::MultiVector<const double> nodal_vector = Core::LinAlg::MultiVector<double>(
      *discretization.node_row_map(), element_multi_vector.NumVectors(), true);
```
or `nodally_averged_vector`?